cluster-autoscaler/core/scaledown/pdb/pdb.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pdb
+
+import (
+	"fmt"
+
+	policyv1 "k8s.io/api/policy/v1"
+)
+
+// PdbRemainingDisruptions stores how many discuptiption is left for pdb.
+type PdbRemainingDisruptions struct {
```

Comment:
I think it would be much more useful to have pod<->pdb matching logic here, and the following API in this struct:

```
// Init the struct with known PDBs
func NewPdbRemainingDisruptions([]*policyv1.PodDisruptionBudget)
// Check all matching disruptions and return true iff disruption is possible
func CanDisrupt([]*apiv1.Pod) bool
// Update all matching disruptions, decrementing remaining disruption count for each pdb
func Update([]*apiv1.Pod)
```

So, scale down logic would create the object before the simulation, use `CanDisrupt` to check if all the pods on a given node are possible to evict, and then finally `Update` if the node satisfies other criteria for scale down.